### ISSUE TITLE
Category Encoders

### ISSUE THREAD

**Issue by wdm0006** [scikit-learn-contrib/scikit-learn-contrib#11]
Request for project inclusion in scikit-learn-contrib

* Project name: category_encoders
* Project description: a collection of transformers for encoding categorical variables in different ways. Handles input/output as either pandas DataFrames or numpy arrays well, though has been primarily used with DataFrames.  Will encode only certain columns, all columns, or infer categorical columns for you.
* Authors: Will McGinnis
* Current repository: [categorical_encoding](https://github.com/wdm0006/categorical_encoding/)
* Requirements:
 * [x] scikit-learn compatible (check_estimator passed)
 * [x] Documentation (guide, API reference, example gallery)
 * [x] Unit tests (coverage: 94%)
 * [x] Python3 compatible
 * [x] PEP8 compliant
 * [x] Continuous integration

**Comment by GaelVaroquaux** [comment]
How is this project positionned compared to sklearn-pandas?

**Comment by wdm0006** [comment]
Should still work fine if you used sklearn-pandas to map a category_encoders transformation to a set of columns.   They don't really do the same thing (though probably there is some internal functionality shared between the two), so I'd consider the two packages complimentary. 

Alternatively, you can just pass in the DataFrame or numpy array to a category_encoders transformer directly and specify `cols=[foo, bar, baz]` where either column labels (for df) or indices (for df or np array) are the columns you want to encode.   Some of the encodings are multivariate, so a per-column mapping type operation would yield different results than using the encoder as intended.

Currently, to support this, we have a requirement for pandas, but in the next major release I'm hoping to soften that from a hard requirement, to a requirement only if you are actually using DataFrames as the primary dataset representation.

**Comment by mblondel** [comment]
Some classes (e.g., polynomial coding) are missing a description in the API reference.

Besides that, this looks like a nice little project. +1 on my side. 

@GaelVaroquaux Did @wdm0006 answer your question?

Any other opinions?

**Comment by jnothman** [comment]
Yes, I agree it looks useful, and that its main problem is the lack of description in each estimator's documentation.

In general (see PEP257), docstrings should start with a short description:
```python
class FooBar:
    """What this class is about

    More detailed description
    """
    do stuff
```

Also, doctests are used for examples in these strings, but it'd be very useful for the unfamiliar user if the output of `enc.transform(X)` was actually shown rather than stored in `numeric_dataset`.

**Comment by wdm0006** [comment]
Thanks all, I'll take a look at the docstrings throughout the project, let me know if there is anything else that looks off.

**Comment by wdm0006** [comment]
@mblondel @GaelVaroquaux @jnothman: I have put in some basic descriptions in the API reference and updated the docstrings to have not only the examples, but the output of the examples.  Further building out the documentation with richer examples is a priority going forward, are there any other points of feedback re: inclusion/exclusion in scikit-learn-contrib?

**Comment by jnothman** [comment]
Generally I am +1 for inclusion, though I still find it hard to identify what the output looks like from the doctests.

**Comment by mblondel** [comment]
+1 on my side too

**Comment by wdm0006** [comment]
Any next steps on my end or are we waiting for more reviewers?

**Comment by mblondel** [comment]
If nobody opposes, I will give you the permissions this week-end.

**Comment by wdm0006** [comment]
Awesome, thanks!

**Comment by amueller** [comment]
I find the documentation now very clear. It could really use some examples on what the different encoders do.

**Comment by mblondel** [comment]
@wdm0006 I added you to the scikit-learn-contrib organization. You should have received the invite. If you don't mind, before/after you transfer the project, I would rename it to `categorical-encoding` (with an hyphen) for consistency with other contrib projects.

**Comment by wdm0006** [comment]
Thanks @mblondel! I've transferred the project over and renamed it to categorical-encoding.

**Comment by mblondel** [comment]
Thanks! Closing.

**Comment by mblondel** [comment]
@wdm0006 As Andreas mentioned, it would be useful to have actual output examples of each transformer.
